File: codeforces/1374/C/1374C.java
import java.io.*;
import java.util.*;
// note the class name and the public identifier may differ from one online judge to the other.
public class Main {
    public static String solucion(int cantidadParentesis, String[] Pparentesis){
        String answer="";
        Stack <String>pila= new Stack<String>();
        String [] parentesis=Pparentesis;
        for(int i=0;i<parentesis.length;i++){
            if(pila.isEmpty()){
                pila.push(parentesis[i]);
            }else{
                if(pila.peek().equals("(")&&parentesis[i].equals(")")){
                    pila.pop();
                }else{
                    pila.push(parentesis[i]);
                }
            }
        }
        if(pila.isEmpty()){
            answer=""+0;
        }else{
            answer=""+pila.size()/2;
        }
        return answer;
    }
public static void main(String [] argv) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int cantidad= Integer.parseInt(br.readLine());
        String[] respuestas= new String[cantidad];
        for(int i=0;i<cantidad;i++){
            int cantidadParentesis= Integer.parseInt(br.readLine());
            String [] parentesis= br.readLine().split("");
            respuestas[i]= solucion(cantidadParentesis, parentesis);
        }
        for(int i=0;i<cantidad;i++){
            System.out.println(respuestas[i]);
        }
    }
}

File: codeforces/1374/C/1374C.kt
package codeforces

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val In = BufferedReader(InputStreamReader(System.`in`))
    val test = In.readLine().toInt()

    tailrec fun sol(l : List<Char>, count : Int, ans : Int) : Int  {
        if(l.isEmpty())
            return ans
        else {
            val h = l.first()
            if (h == '(')
                return sol(l.drop(1), count + 1, ans)
            else {
                if (count - 1 == -1)
                   return sol(l.drop(1), 0, ans + 1)
                else
                   return  sol(l.drop(1), count - 1, ans)
            }
        }
    }

    for(i in 1..test) {
        In.readLine()
        val paren = In.readLine().toList()
        println(sol(paren, 0, 0))
    }
}